Added a new post outlining my proposal for CSC 596 (#22)
compusciencing/compusciencing.github.io: _posts/7-31-2019--INTROPOST.md

---
layout: post
title: INTROPOST
author: Normandy Rivver
categories: ['Missouri State University', 'Computer Science']

> tags: ['CSC 596', 'robotics', 'robotics-game', 'transformable-robot-game']

Reviewer: ```suggestion
tags: ['csc596', 'robotics', 'robotics-game', 'transformable-robot-game']
```

Author: Completed

> 

Reviewer: Delete this blank line.

Author: I assume you mean the blank line above the ---- and below the ----

---

> Working with Dr. Clark, my research project will involve learning and implementing various JavaScript libraries to create an interactive game teaching user’s about robotics.  The main library I will use is a 2D game framework called [phaser.io](https://phaser.io/).  With phaser.io I will achieve certain project goals, such as building a robotic car with extendable wheel-legs, allowing users to create and configure their own unique robotic cars, and a health bar for keeping track of a vehicle's health during game play.  To complete some of these project goals, the project will rely on both [Matter.js](http://brm.io/matter-js/), a 2D physics engine for the web, and [Svelte.js](https://svelte.dev/), a library for building user interfaces.  This is not an exhaustive list of libraries I will use and only serves as a guide to current project goals.  The goal is to create an interactive environment allowing users to build a robot with transformable wheels to traverse a rough terrain. Terrains may be pre-built or generated at random; since specifics are lacking on this, I will leave the option of doing either or both open.  Another slight unknown is how the user interface will play out.  We have discussed no specifics regarding how a user interface should/will be used but the goal is to utilize best practices as much as possible when designing the user interface.  I consider these unknowns as part of the project since one cannot always plan out an approach which leads to project completion before beginning the actual project.  I will deal with these gaps when confronted during the project.  Ultimately, we would like the user to specify the robot design from a slider menu and the terrain the robot will encounter.  The environment will be a side scrolling game allowing the user to engage significantly longer with the aim of keeping the users high health score for a given session on a given terrain displayed to the user.

Reviewer: ```suggestion
Working with Dr. Clark, my research project will involve learning and implementing various JavaScript libraries to create an interactive game teaching user’s about robotics.  The main library I will use is a 2D game framework called [phaser.io](https://phaser.io/).  With phaser.io I will achieve certain project goals, such as building a robotic car with extendable wheel-legs, allowing users to create and configure their own unique robotic cars, and a health bar for keeping track of a vehicle's health during game play.  To complete some of these project goals, the project will rely on both [Matter.js](http://brm.io/matter-js/), a 2D physics engine for the web, and [Svelte.js](https://svelte.dev/), a library for building user interfaces.  This is not an exhaustive list of libraries I will use and only serves as a guide to current project goals.  The goal is to create an interactive environment allowing users to build a robot with transformable wheels to traverse a rough terrain. Terrains may be pre-built or generated at random; since specifics are lacking on this, I will leave the option of doing either or both open.  Another slight unknown is how the user interface will play out.  We have discussed no specifics regarding how a user interface should/will be used but the goal is to utilize best practices as much as possible when designing the user interface.  I consider these unknowns as part of the project since one cannot always plan out an approach which leads to project completion before beginning the actual project.  I will deal with these gaps when confronted during the project.  Ultimately, we would like the user to specify the robot design from a slider menu and the terrain the robot will encounter.  The environment will be a side scrolling game allowing the user to engage significantly longer to keep the users high health score for a session on a terrain displayed to the user.
```

Reviewer: You could also break that into multiple paragraphs.

Author: Broken into paragraphs for easier readability.

> Weekly updates  are required by Dr. Clark with a template already provided.  A mid-term self-evaluation and a final project summary make up the additional requirements.  We are also scheduled to meet once a week but have not selected a date/time yet.

Reviewer: ```suggestion
Dr. Clark requires weekly updates using a provided template.  A mid-term self-evaluation and a final project summary make up the additional requirements.  We are also scheduled to meet once a week but have not selected a date/time yet.
```
